src/cursor/abstract_cursor.ts

```diff
@@ -620,94 +629,111 @@ export abstract class AbstractCursor<
 
     executeOperation(this.topology, getMoreOperation, callback);
   }
-}
-
-function nextDocument<T>(cursor: AbstractCursor): T | null | undefined {
-  if (cursor[kDocuments] == null || !cursor[kDocuments].length) {
-    return null;
-  }
-
-  const doc = cursor[kDocuments].shift();
-  if (doc) {
-    const transform = cursor[kTransform];
-    if (transform) {
-      return transform(doc) as T;
-    }
-
-    return doc;
-  }
-
-  return null;
-}
-
-function next<T>(cursor: AbstractCursor, blocking: boolean, callback: Callback<T | null>): void {
-  const cursorId = cursor[kId];
-  if (cursor.closed) {
-    return callback(undefined, null);
-  }
 
-  if (cursor[kDocuments] && cursor[kDocuments].length) {
-    callback(undefined, nextDocument(cursor));
-    return;
-  }
-
-  if (cursorId == null) {
-    // All cursors must operate within a session, one must be made implicitly if not explicitly provided
-    if (cursor[kSession] == null) {
-      if (cursor[kTopology].shouldCheckForSessionSupport()) {
-        return cursor[kTopology].selectServer(ReadPreference.primaryPreferred, {}, err => {
+  /**
+   * @internal
+   *
+   * This function is exposed for the unified test runner's createChangeStream
+   * operation.  We cannot refactor to use the abstract _initialize method without
+   * a significant refactor.
+   */
+  __initialize(callback: Callback<TSchema | null>): void {
+    if (this[kSession] == null) {
+      if (this[kTopology].shouldCheckForSessionSupport()) {
+        return this[kTopology].selectServer(ReadPreference.primaryPreferred, {}, err => {
           if (err) return callback(err);
-          return next(cursor, blocking, callback);
+          return this.__initialize(callback);
         });
-      } else if (cursor[kTopology].hasSessionSupport()) {
-        cursor[kSession] = cursor[kTopology].startSession({ owner: cursor, explicit: false });
+      } else if (this[kTopology].hasSessionSupport()) {
+        this[kSession] = this[kTopology].startSession({ owner: this, explicit: false });
       }
```

Comment:
A bit of scope creep but I think it's an easy win, if we make ExecutionResult generic now, and then also _initialize generic, we should be able to pass through and interface that has the minimal properties used in this code here, keeping it correct.

```
this._initialize<
      | { cursor: { id: Long | number; ns?: string; firstBatch: TSchema[] } }
      | { cursorId: number | Long; documents: TSchema[] } // old way, can be removed
    >(this[kSession], (err, state) => {
```

Reply:
seconding the follow up thought

Reply:
Good catch on the "raw" case, yea let's do this later.

Reply:
just to confirm, are we thinking of doing this in NODE-3286? if so, can we capture both of your comments there? (both Neal's suggestion and bailey's note about raw)

Reply:
Well so maybe my code block over did it, I was just thinking right now the `state.response` type is Document, if we type it to be:
```ts
type x = { cursor: { id: Long | number; ns?: string; firstBatch: unknown[] } }
```
We can at least get assertions on `id` and `ns` but the schema isn't actually relevant to this function at all. So my goal wasn't to make the schema go lower but instead type our expectation of the response shape (for example the potential Long or number for id). Seems like a small win

Reply:
Or another option, could be to type put the annotation on the variable we make:
```
const response: Record<string, unknown> = state.response;
// This would make sure our assertions about each of the properties `id` and `ns` is type safe
```

Reply:
Unless you feel strongly about this particular fix, I'm going leave it as-is.  I think that rather than retroactively typing this function with a type that matches how its used, we should try and fix this generally in the follow up ticket.